# Module 3 Lab 3: Textbook Problem 2.2


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("bridges.csv")


## Plausibility Assumption

I first analyzed the table provided in Problem 2.2. There are 35 bridges, and length data are missing for 14 of them, which is about 40%. Therefore, I see a problem with the amount of missing data. Only one bridge has a length of 4558, which may still be possible. The Span categories do not show a clear correlation with the Length values, especially for Short and Medium.

I assume that bridge length is measured in feet. For this analysis, I consider positive bridge lengths up to 1600 feet plausible. Values greater than 1600 feet will be flagged as unusually large and investigated, but they will not automatically be considered incorrect.


In [3]:
df.head()


,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
0,E1,M,3.0,1818,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
1,E2,A,25.0,1819,HIGHWAY,1037.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
2,E3,A,39.0,1829,AQUEDUCT,NaN,1.0,N,THROUGH,WOOD,NaN,S,WOOD
3,E5,A,29.0,1837,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
4,E6,M,23.0,1838,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,NaN,S,WOOD


In [4]:
df.dtypes


Id              str
river           str
location    float64
erected       int64
purpose         str
length      float64
lanes       float64
clear-g         str
t-or-d          str
material        str
span            str
rel-l           str
type            str
dtype: object

In [5]:
df["length"].describe()

count      81.000000
mean     1567.469136
std       747.491523
min       804.000000
25%      1000.000000
50%      1300.000000
75%      2000.000000
max      4558.000000
Name: length, dtype: float64

In [6]:
print("Total bridges:", len(df))
print("Missing length values:", df["length"].isna().sum())

Total bridges: 108
Missing length values: 27


In [7]:
flagged = df[(df["length"] <= 0) | (df["length"] > 1600)]
flagged

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
31,E34,O,41.0,1888,RR,4558.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
41,E42,M,9.0,1895,HIGHWAY,2367.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
44,E46,A,37.0,1897,RR,4000.0,2.0,G,DECK,STEEL,LONG,F,SIMPLE-T
45,E45,M,14.0,1897,RR,2264.0,NaN,G,THROUGH,STEEL,NaN,F,SIMPLE-T
46,E47,M,15.0,1898,RR,2000.0,2.0,G,THROUGH,STEEL,MEDIUM,S,SIMPLE-T
48,E48,A,38.0,1900,HIGHWAY,2000.0,2.0,G,THROUGH,STEEL,MEDIUM,F,SIMPLE-T
50,E49,A,34.0,1902,HIGHWAY,1850.0,2.0,G,THROUGH,STEEL,MEDIUM,F,CANTILEV
52,E87,A,35.0,1903,RR,3000.0,2.0,G,THROUGH,STEEL,MEDIUM,S,SIMPLE-T
60,E55,A,36.0,1909,HIGHWAY,1730.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
61,E57,O,49.0,1910,RR,1620.0,2.0,G,THROUGH,STEEL,LONG,F,CANTILEV


## Review of My Initial Assumption
My initial assumption was based on the textbook table, but the complete CSV showed a broader distribution. Therefore, I need to review my initial assumption after investigating the flagged values.

In [8]:

print("Number flagged:", len(flagged))
print("Percent of available lengths flagged:", round(len(flagged) / df["length"].count() * 100, 1), "%")
print("Flagged length range:", flagged["length"].min(), "to", flagged["length"].max())

Number flagged: 30
Percent of available lengths flagged: 37.0 %
Flagged length range: 1620.0 to 4558.0


## Final Conclusion and Handling

My initial assumption of 1600 feet was too restrictive because I did not have all the data. Now I can see that it flagged 30 of the 81 available length values, or about 37%. The summary statistics also show that the median length is 1300 feet, the 75th percentile is 2000 feet, and the maximum is 4558 feet. This means that values greater than 1600 are not unusual enough, and we cannot consider them outliers only because they exceed my initial limit.

After reviewing the flagged rows, I do not think these values should be removed or changed. The largest value, 4558, is unusually high, but it may be reasonable because it is a railroad bridge made of steel and is classified as Long, which supports keeping the value. I would keep the flagged values and document them rather than correct or remove them without additional information.